### PR TITLE
Fix parsing of MusicBrainz XML response

### PR DESCRIPTION
**Pull request by uklotzde**

### CODE REVIEW

**Review by daschuer — APPROVED** [review]
LGTM, does this really "Fix" something?
Is there a need for a manual test?

**Comment by uklotzde** [comment]
> LGTM, does this really "Fix" something?
> Is there a need for a manual test?

Yes, all "release" elements within the "release-list" where skipped, because the previous `name` from the outer context was used for comparison. I wondered why I didn't get **any** results when querying MusicBrainz. Very hard to catch bug!

File: src/musicbrainz/musicbrainzxml.cpp [hunk 1]
```diff
@@ -393,6 +393,7 @@ QPair<QList<TrackRelease>, bool> readElementRecording(QXmlStreamReader& reader)
                 while (continueReading(reader)) {
                     const QXmlStreamReader::TokenType type = reader.readNext();
                     if (type == QXmlStreamReader::StartElement) {
+                        const QStringRef name = reader.name();
```
Comment (Be-ing) [inline src/musicbrainz/musicbrainzxml.cpp:396]: You are defining a variable with the same name as a different variable in the surrounding scope. This is a recipie for very hard to catch bugs like #1447.

Comment (uklotzde) [reply src/musicbrainz/musicbrainzxml.cpp:396]: I am not going to touch this code again. Manual parsing is a pita.

Comment (uklotzde) [reply src/musicbrainz/musicbrainzxml.cpp:396]: The only alternative is to replace it with the call `reader.name()` literally.

Comment (uklotzde) [reply src/musicbrainz/musicbrainzxml.cpp:396]: I am not going to split the code into separate functions.

Comment (Be-ing) [reply src/musicbrainz/musicbrainzxml.cpp:396]: I am not suggesting splitting the code into separate functions. But you can rename one of the variables.

Comment (uklotzde) [reply src/musicbrainz/musicbrainzxml.cpp:396]: Giving them arbitrary names is not an option for me.

Comment (Holzhaus) [reply src/musicbrainz/musicbrainzxml.cpp:396]: `readerName` wouldn't be arbitrary IMHO.

Comment (Be-ing) [reply src/musicbrainz/musicbrainzxml.cpp:396]: How about making the outer variable `startElementName` and the inner one `releaseListName`?

Comment (uklotzde) [reply src/musicbrainz/musicbrainzxml.cpp:396]: I just fixed the bug. Everything else would add unnecessary noise to this PR. We are expecting one or more values, but we don't know in advance what's in there. The proposed names don't fit.

The variable is just an alias of what's currently stored in reader.name(). Using `readerName` doesn't twice doesn't make it better. I will not change anything here unless it is a real improvement.

Comment (Be-ing) [reply src/musicbrainz/musicbrainzxml.cpp:396]: This is a highly error prone solution. This is not hypothetical. I remember it took me 8 hours to solve #1447. Even if the variable names are not that descriptive, IMO we should never in any circumstances define a variable with the same name as the surrounding scope. It is surprising that compilers even allow this.

Comment (Be-ing) [reply src/musicbrainz/musicbrainzxml.cpp:396]: You could even call it `nameInner` for lack of a better term if you want. Just not the same name as the outer scope.

Comment (uklotzde) [reply src/musicbrainz/musicbrainzxml.cpp]: Now it is consistent with the remaining code, i.e. no variables at all.

**Review by Be-ing — APPROVED** [review]
Waiting for CI

File: src/musicbrainz/musicbrainzxml.cpp (hunk 1 above)
Comment (Be-ing) [reply src/musicbrainz/musicbrainzxml.cpp]: Sure, that works too.

**Comment by Be-ing** [comment]
@Holzhaus do you know of any tools we could use in CI that detect shadowing variable names?

**Comment by uklotzde** [comment]
Spurious test failure on Windows again.

**Comment by uklotzde** [comment]
> @Holzhaus do you know of any tools we could use in CI that detect shadowing variable names?

https://github.com/uklotzde/mixxx/actions/runs/416381873

Though I don't expect that this build gets far ;)

**Pull request merged**